maple2latex/src/translator.py

```diff
@@ -68,6 +72,126 @@ def __init__(self, inp):
+        eq.front = translate(eq.fields["front"])
+
+        equation = eq.lhs + "\n  = "
+
+        if eq.factor == "1":
+            eq.factor = ""
+
+        metadata = dict()
+
+        # translates the Maple information (with spacing)
+        if eq.eq_type == "series":
+            eq.general = translate(eq.general[0])
+
+            if eq.factor != "":
+                equation += eq.factor + " "
+
+            elif eq.front != "":
+                equation += eq.front + "+"
+
+            equation += "\\sum_{k=0}^\\infty "
+
+            if eq.category == "power series":
+                equation += eq.general
+            elif eq.category == "asymptotic series":  # make sure to fix asymptotic series
+                equation += "(" + eq.general + ")"
+
+        elif eq.eq_type == "contfrac":
+            forms = list()
+
+            # substitution handling
+            if len(parse_brackets(eq.general[0])) > 1:
+                forms = parse_brackets(eq.general[0])
+
+            if len(eq.general) == 2 or forms:
+                if not forms:  # forms is empty
+                    for form in eq.general:
+                        forms += parse_brackets(form)
+
+                metadata["substitution"] = ','.join(perform_substitution(eq, forms))
+                pieces = ["s_m", "1"]
+
+            else:
+                pieces = parse_brackets(eq.general[0])[0]
+
+            if eq.begin != "":
+                eq.begin = parse_brackets(eq.begin)
+
+            start = 1  # in case the value of start isn't assigned
+
+            # add terms before general
+            if eq.front != "":
+                equation += eq.front + "+"
+                start = 1
+
+            if eq.begin != "":
+                for piece in eq.begin:
+                    equation += make_frac(piece) + " \\subplus "
+                    start += 1
+
+            if eq.factor != "":
+                if eq.factor == "-1":
+                    equation += "-"
+                else:
+                    equation += eq.factor + " "
+
+            # trim unnecessary parentheses
+            for i, element in enumerate(pieces):
+                pieces[i] = trim_parens(translate(element))
+
+            if pieces != ["0", "1"]:
+                equation += "\\CFK{m}{" + str(start) + "}{\\infty}@@{" + pieces[0] + "}{" + pieces[1] + "}"
+            else:
+                equation += "\\dots"
+
+        metadata["constraint"] = translate(eq.constraints)
+        metadata["category"] = eq.category
+
+        return cls(eq.label, replace_strings(equation, SPECIAL), metadata)
+
+    @staticmethod
+    def get_sortable_label(equation):
+        if equation.label == "x.x.x":
+            return [100, 100, 100]  # should be three very large numbers
+
+        label = copy.copy(equation.label)
+        for i, ch in enumerate(ascii_lowercase):
+            if label[-1] == ch:
+                label = label[:-1] + "." + str(i + 1)
+
+        if label[-1] == ".":
```

Comment:
If the point of this is to remove any occurrences of empty strings when you `split` and `map` the string lower down, consider simply changing said `split` and `map` to `map(int, filter(lambda x: x != "", label.split(".")))`.